crates/rover-client/src/operations/subgraph/fetch_all/runner.rs

```diff
@@ -0,0 +1,136 @@
+use graphql_client::*;
+
+use crate::blocking::StudioClient;
+use crate::operations::config::is_federated::{self, IsFederatedInput};
+use crate::RoverClientError;
+
+use super::types::*;
+
+#[derive(GraphQLQuery)]
+// The paths are relative to the directory where your `Cargo.toml` is located.
+// Both json and the GraphQL schema language are supported as sources for the schema
+#[graphql(
+    query_path = "src/operations/subgraph/fetch_all/fetch_all_query.graphql",
+    schema_path = ".schema/schema.graphql",
+    response_derives = "Eq, PartialEq, Debug, Serialize, Deserialize",
+    deprecated = "warn"
+)]
+/// This struct is used to generate the module containing `Variables` and
+/// `ResponseData` structs.
+/// Snake case of this name is the mod name. i.e. subgraph_fetch_all_query
+pub(crate) struct SubgraphFetchAllQuery;
+
+/// For a given graph return all of its subgraphs as a list
+pub fn run(
+    input: SubgraphFetchAllInput,
+    client: &StudioClient,
+) -> Result<Vec<Subgraph>, RoverClientError> {
+    // This response is used to check whether the current graph is federated.
+    let is_federated = is_federated::run(
+        IsFederatedInput {
+            graph_ref: input.graph_ref.clone(),
+        },
+        client,
+    )?;
+    if !is_federated {
+        return Err(RoverClientError::ExpectedFederatedGraph {
+            graph_ref: input.graph_ref,
+            can_operation_convert: false,
+        });
+    }
```

Comment:
flagging as what might be removable or at least in conflict with some recent changes from glasser

Reply:
Oh yer that's just a bad rebase, thanks for flagging :)